third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc

```diff
@@ -962,24 +960,37 @@ static String MakeReplayEventType(const String& eventTypeRaw,
   return builder.ToString();
 }
 
+static int gIsEventInFlight = 0;
+
 void ReplayNotifyBeforeEvent(const String& eventName, 
-                      EventTarget* eventTarget,
-                      bool isCallback) {
-  String qualifiedEventName =
+                             EventTarget* eventTarget,
+                             bool isCallback) {
+  String replayEventType =
       MakeReplayEventType(eventName, eventTarget, isCallback);
-  // Disabled, see https://linear.app/replay/issue/RUN-1206
-  //recordreplay::OnEvent(qualifiedEventName.Ascii().c_str(), true);
-  (void)qualifiedEventName;
+
+  // Disabled by default, see https://linear.app/replay/issue/RUN-1251
+  if (recordreplay::IsRecordingOrReplaying() &&
+      !recordreplay::FeatureEnabled("disable-collect-events")) {
```

Comment:
```
  if (recordreplay::IsRecordingOrReplaying() &&
      !recordreplay::FeatureEnabled("disable-collect-events")) {
```

Not a big deal, but it is typically easier to think in terms of feature flags enabling features as opposed to double negatives...

```
  if (recordreplay::IsRecordingOrReplaying() && recordreplay::FeatureEnabled("collect-events")) {
```

To your point about the feature being disabled by default. It's totally fine for `collect-events` to be `false` by default :) Feels good to turn things on 😉

Reply:
> it is typically easier to think in terms of feature flags enabling features as opposed to double negatives...

No one wants double negatives, but that design is not really up to me :cry:. It was done by @bhackett1024 and he does not want to "disabled defaults". Hence, we cannot actually change the defaults. I am following his lead on this. Several other features were added in the past with a `no-` and `disabled-` prefix for the same reason.

Regardless, Brian already agreed that we need a better and more general-purpose API [here](https://github.com/replayio/backend/pull/7030#pullrequestreview-1283354760), so I filed a follow up issue for this [here](https://linear.app/replay/issue/RUN-1263/feature-flag-api-for-runtimes).

Reply:
@jasonLaster [here](https://github.com/replayio/chromium-v8/blob/31707b5e1933d9ca33d3b95a9f82f610698c0da2/src/api/api.cc#L10959) is the list of available flags.